proxy/http2/HPACK.cc

```diff
@@ -227,19 +223,18 @@ hpack_parse_field_type(uint8_t ftype)
 namespace HpackStaticTable
 {
   HpackLookupResult
-  lookup(const char *name, int name_len, const char *value, int value_len)
+  lookup(const HpackHeaderField &header)
   {
     HpackLookupResult result;
 
     for (unsigned int index = 1; index < TS_HPACK_STATIC_TABLE_ENTRY_NUM; ++index) {
-      const char *table_name  = STATIC_TABLE[index].name;
-      int table_name_len      = STATIC_TABLE[index].name_size;
-      const char *table_value = STATIC_TABLE[index].value;
-      int table_value_len     = STATIC_TABLE[index].value_size;
+      std::string_view name  = STATIC_TABLE[index].name;
+      std::string_view value = STATIC_TABLE[index].value;
 
+      // TODO: replace `strcasecmp` with `memcmp`
       // Check whether name (and value) are matched
-      if (ptr_len_casecmp(name, name_len, table_name, table_name_len) == 0) {
-        if ((value_len == table_value_len) && (memcmp(value, table_value, value_len) == 0)) {
+      if (strcasecmp(header.name, name) == 0) {
+        if (header.value.size() == value.size() && memcmp(header.value.data(), value.data(), value.size()) == 0) {
```

Comment:
I think `memcmp` is overloaded the same way as `strcasecmp`, e.g. you could do
```
if (memcmp(header.value, value) == 0) {
```
However, the equality operator also uses `memcmp` so both of these are equivalent to
```
if (header.value == value) {
```

Reply:
I completely forgot about `memcmp` in `TextView.h`.  Will call it.

Reply:
It looks like the equality operator calls `compare` in some implementations (- e.g. libc++ ). `compare` seems bit slow, so I'll go with `memcmp` in `TextView.h`

Reply:
Done.